src/VirtualClient/VirtualClient.Actions/3DMark/ThreeDMarkMetricsParser.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection.Metadata;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Xml.Linq;
+    using System.Xml.Serialization;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Parser for the NTttcp workload.
+    /// </summary>
+    public class ThreeDMarkMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// Parser for the 3DMark workload
+        /// </summary>
+        /// <param name="rawText">The raw text from the 3DMark export process.</param>
+        /// <param name="definition">The 3dmark definition.</param>
+        public ThreeDMarkMetricsParser(string rawText, string definition)
+            : base(rawText)
+        {
+            this.Defintion = definition;
+        }
+
+        private string Defintion { get; set; }
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            IList<Metric> metrics = new List<Metric>();
+            try
+            {
+                if (this.Defintion == "custom_TSGT1.3dmdef")
+                {
+                    metrics.Add(new Metric("timespy.graphics.1 [fps]", this.ParseXMLTag("TimeSpyPerformanceGraphicsTest1"), "fps", MetricRelativity.HigherIsBetter));

```

Comment:
I'm just curious, where did the names TimeSpyPerformanceGraphicsTest1, TimeSpyPerformanceGraphicsTest2, TimeSpyPerformanceCpuSection2 come from, is it robust enough so that it will handle TimeSpyPerformanceCpuSection1 for example? Or the names are fixed? I would rather parse every Xml element. I can make the change if you prefer.